tests/functional/test_regions.py

```diff
@@ -545,3 +551,58 @@ def test_endpoint_resolver_knows_its_datasource(patched_session, is_builtin):
     with mock.patch.object(loader, 'is_builtin_path', return_value=is_builtin):
         resolver = session._get_internal_component('endpoint_resolver')
         assert resolver.uses_builtin_data == is_builtin
+
+
+class TestCustomEndpointsFile(BaseSessionTest):
+    def setUp(self):
+        super().setUp()
+        self.tempdir = tempfile.TemporaryDirectory()
+        self.environ['AWS_DATA_PATH'] = self.tempdir.name
+        self.session = Session()
+        self.session.set_config_variable(
+            'credentials_file', 'noexist/foo/botocore'
+        )
+        self.session.config_filename = 'no-exist-foo'
```

Comment:
Discussed offline. The point of this was to mimic everything we're doing in `BaseSessionTest` and `create_session`. We needed to create our own session because the method uses a cached `Loader` instance which does not have the custom data path attached to it. However, since there are other tests that check that the env vars work as expected, we're going to modify `create_session` instead to pass a custom loader